src/controllers/RythmController.ts

```diff
@@ -46,8 +47,16 @@ const handleEvent = async (message: Message): Promise<void> => {
 const songStartedLongAgo = (song: Song): boolean => {
     const now = Date.now();
     const start = song.time ?? now;
-    // TODO implement stub
-    return true;
+    const length = song.length;
+    if ((now - start) >= 30000) {
+        Log.debug( `${song.name} started over 30 seconds`);
+        return true;
+    } else if (now > start + (length/2)) {
```

Comment:
I really should add this to the linter, but could we have proper spacing between language terms?

This would look like `length / 2` here!